proposals/async-method-builders.md

```diff
@@ -46,23 +46,70 @@ namespace System.Runtime.CompilerServices
     /// Indicates the type of the async method builder that should be used by a language compiler to
     /// build the attributed method.
     /// </summary>
-    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Module, Inherited = false, AllowMultiple = false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Module, Inherited = false, AllowMultiple = true)]
     public sealed class AsyncMethodBuilderOverrideAttribute : Attribute
     {
         /// <summary>Initializes the <see cref="AsyncMethodBuilderOverrideAttribute"/>.</summary>
         /// <param name="builderType">The <see cref="Type"/> of the associated builder.</param>
         public AsyncMethodBuilderOverrideAttribute(Type builderType) => BuilderType = builderType;
 
-        // for scoped application (use property for targetReturnType? have compiler check that it's provided)
-        public AsyncMethodBuilderOverrideAttribute(Type builderType, Type targetReturnType) => ...
-
         /// <summary>Gets the <see cref="Type"/> of the associated builder.</summary>
         public Type BuilderType { get; }
     }
 }
 ```
 
-In the C# compiler, prefer the attribute on the method when determining what builder to use over the one defined on the type.  For example, today if a method is defined as:
+The attribute can be applied on methods, types and modules.
+
+Example of usage on a method:  
+```C#
+[AsyncMethodBuilderOverride(typeof(PoolingAsyncValueTaskMethodBuilder<int>))] // new, referring to some custom builder type
+static async ValueTask<int> ExampleAsync() { ... }
+```
+
+A developer that wants to using a specific custom builder for all of their methods can do so by putting the relevant attribute on each method.  
+Example of usage on module:  
+```C#
+[module: AsyncMethodBuilderOverride(typeof(PoolingAsyncValueTaskMethodBuilder))]
+[module: AsyncMethodBuilderOverride(typeof(PoolingAsyncValueTaskMethodBuilder<>))]
+
+class MyClass
+{
+    public async ValueTask Method1Async() { ... } // would use PoolingAsyncValueTaskMethodBuilder
+    public async ValueTask<int> Method2Async() { ... } // would use PoolingAsyncValueTaskMethodBuilder<int>
+    public async ValueTask<string> Method3Async() { ... } // would use PoolingAsyncValueTaskMethodBuilder<string>
```

Comment:
Consider adding a counter example here using a task-like type that does not hit the custom builder. Essentially demonstrating the fallback / default case.

Reply:
Added example of picking inner scope and also falling back to non-override behavior.